newt/toolchain/compiler.go

```diff
@@ -292,10 +293,20 @@ func loadFlags(yc ycfg.YCfg, settings *cfgv.Settings, key string) []string {
 		}
 	}
 
+	if cfg != nil {
+		for i, flag := range flags {
+			for _, c := range cfg.Names() {
+				if strings.Contains(flag, "$("+c+")") && cfg.Exists(c) {
+					flags[i] = strings.ReplaceAll(flag, "$("+c+")", cfg.Get(c))
```

Comment:
no need to iterate all syscfg names to find matches, use `ReplaceAllStringFunc` instead